src/app.rs

```diff
@@ -2250,8 +2261,15 @@ impl eframe::App for ProfApp {
         eframe::set_value(storage, eframe::APP_KEY, self);
     }
 
+    fn post_rendering(&mut self, _screen_size_px: [u32; 2], frame: &eframe::Frame) {
+        // this is inspired by the Egui screenshot example
+        if let Some(screenshot) = frame.screenshot() {
+            self.screenshot = Some(screenshot);
+        }
+    }
```

Comment:
[ref](https://github.com/hacknus/egui/blob/4e7f5870a6d77afb19cbd0ebab1c2c8a712bb41a/crates/eframe/src/epi/mod.rs#L782C1-L783C40):

    /// Request the current frame's pixel data. Needs to be retrieved by calling [`Frame::screenshot`]
    /// during [`App::post_rendering`].